resources/src/operation/mod.rs

```diff
@@ -0,0 +1,220 @@
+use amount::Amount;
+use asset::AssetIdentifier;
+use serde::{de, Deserialize, Deserializer};
+use offer::PriceRatio;
+mod create_account;
+use self::create_account::CreateAccountFields;
+
+/// Operations are objects that represent a desired change to the ledger: payments, offers to
+/// exchange currency, changes made to account options, etc. Operations are submitted to the
+/// Stellar network grouped in a Transaction.
+
+#[derive(Debug, Serialize)]
+pub struct Operation {
+    id: i64,
+    paging_token: String,
+    type_i: u32,
+    operation_detail: OperationDetail,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub enum OperationDetail {
+    /// Create Account operation represents a new account creation.
+    CreateAccount(CreateAccountFields),
+    /// A payment operation represents a payment from one account to another. This payment can be
+    /// either a simple native asset payment or a fiat asset payment.
+    Payment(PaymentFields),
+}
+
+#[derive(Debug, Deserialize)]
+pub struct IntermediateOperation {
+    id: i64,
+    paging_token: String,
+    #[serde(rename = "type")] operation_type: String,
+    type_i: u32,
+    account: Option<String>,
+    funder: Option<String>,
+    starting_balance: Option<Amount>,
+}
+
+impl Operation {
+    /// The canonical id of this operation, suitable for use as the :id parameter for url templates
+    /// that require an operation’s ID.
+    pub fn id(&self) -> i64 {
+        self.id
+    }
+
+    /// A paging token suitable for use as a cursor parameter.
+    pub fn paging_token(&self) -> &String {
+        &self.paging_token
+    }
+
+    /// Specifies the type of operation, See “Types” section below for reference.
+    pub fn type_i(&self) -> u32 {
+        self.type_i
+    }
+
+    /// Returns true if the operation is a create_account operation
+    pub fn is_create_account(&self) -> bool {
+        match self.operation_detail {
```

Comment:
It's up to you but I sometimes like the `if let` syntax in these cases

```rust
if let OperationDetail::CreateAccount(_) = self.operation_detail() {
  true
} else {
  false
}
```

It's an extra line but I find the form a little more eloquent. And if we can shorten the words:

```rust
if let Detail::CreateAccount(_) = self.detail() {
  true
} else {
  false
}
```